### ISSUE TITLE
Router mode should use vue-router

### ISSUE THREAD

**Issue by zsh2401** [amirkian007/vue-awesome-sidebar#5]
I have read a tiny part of source of this project, then I found that it seems to just change hash to route. I suggest that use the useRouter to get the router instance and call router.push, so that this library can adapt to more routing methods such as memory and non-hash

**Comment by amirkian007** [comment]
hi @zsh2401 thanks for opening the issue.
using useRouter requires `vue-router` to be added as a dependency , this means user has to install an extra dependency even if not using vue-router at all.
which i don't think is ideal since this library is made with 0 dependencies . 
I think the better method is to  disable default vue-router functionality with `vueRouterEnabel="false"` and  handle routing with event listener like `@item-click(MenuItem)` , which i think is the better approach. slots is another Alternative. whats your opinion?

**Comment by zsh2401** [comment]
I thought we can add vue-router as peer dependency and dynamically check whether users have installed it.

**Comment by zsh2401** [comment]
Other component libraries that contains sidebar component and provide router mode all uses the vue-router. They consider users who is enabling router mode must installed the vue-router

**Comment by zsh2401** [comment]
Thank you for your jobs, again! Your library has been imported to a business project in my company.

**Comment by amirkian007** [comment]
> Other component libraries that contains sidebar component and provide router mode all uses the vue-router. They consider users who is enabling router mode must installed the vue-router

Yeah I need to check them out and figure out a solution when I have the time

**Comment by amirkian007** [comment]
> Thank you for your jobs, again! Your library has been imported to a business project in my company.

I Am very HAPPY to hear that :))) !!!
Thanks again for using this library and I will keep improving and adding new futures to this library soon!
